core/lib/Thelia/Action/Product.php

```diff
@@ -100,6 +121,451 @@ public function create(ProductCreateEvent $event)
         $event->setProduct($product);
     }
 
+    /*******************
+     * CLONING PROCESS *
+     *******************
+     *
+     * Vars prefixes :
+     * op = Original Product
+     * cp = Clone Product
+     */
+
+    /**
+     * @param ProductCloneEvent $event
+     */
+    public function cloneProduct(ProductCloneEvent $event)
+    {
+        // Get important datas
+        $lang = $event->getLang();
+        $opId = $event->getProductId();
+
+        $op = ProductQuery::create()
+            ->findOneById($opId);
+
+        $opDefaultI18n = ProductI18nQuery::create()
+            ->filterByLocale($lang, Criteria::EQUAL)
+            ->findOneById($opId);
+
+        $opDefaultPrice = ProductPriceQuery::create()
+            ->findOneByProductSaleElementsId($op->getDefaultSaleElements()->getId());
+
+        $opPSEs = ProductSaleElementsQuery::create()
+            ->orderByIsDefault(Criteria::DESC)
+            ->findByProductId($opId);
+
+        // Cloning process
+
+        $cp = $this->createClone($op, $opDefaultI18n, $opDefaultPrice, $event);
+
+        $dispatcher = $event->getDispatcher();
+
+        $cp = $this->updateClone($cp, $op, $opDefaultI18n, $opDefaultPrice, $dispatcher);
+
+        $cpPSEId = $cp->getDefaultSaleElements()->getId();
+
+        $this->setCloneFiles($opId, $cp, $dispatcher);
+
+        // PSEs handling
+        foreach ($opPSEs as $key => $opPSE) {
+            if (!$opPSE->getIsDefault()) {
+                $currencyId = ProductPriceQuery::create()
+                    ->findOneByProductSaleElementsId($opPSE->getId())
+                    ->getCurrencyId();
+
+                $cpPSEId = $this->createClonePSE($cp, $op, $opPSE, $key, $currencyId, $dispatcher);
+            }
+
+            $this->updateClonePSE($cp, $cpPSEId, $op, $opPSE, $key, $dispatcher);
+
+            $this->setCloneAttributeCombination($opPSE->getId(), $cpPSEId);
```

Comment:
Managing AttributeCombination is useless because when you create a new PSE, the listener will create them for you.